official/utils/arg_parsers/parsers.py

```diff
@@ -0,0 +1,208 @@
+
+  Args:
+    add_help: Create the "--help" flag. False if class instance is a parent.
+    train_epochs: Create a flag to specify the number of training epochs.
+    epochs_per_eval: Create a flag to specify the frequency of testing.
+    batch_size: Create a flag to specify the batch size.
+  """
+
+  def __init__(self, add_help=False, train_epochs=False, epochs_per_eval=False,
+               batch_size=False):
+    super(SupervisedParser, self).__init__(add_help=add_help)
+
+    if train_epochs:
+      self.add_argument(
+          "--train_epochs", "-te", type=int, default=1,
+          help="[default: %(default)s] The number of epochs used to train.",
+          metavar="<TE>"
+      )
+
+    if epochs_per_eval:
+      self.add_argument(
+          "--epochs_per_eval", "-epe", type=int, default=1,
+          help="[default: %(default)s] The number of training epochs to run "
+               "between evaluations.",
+          metavar="<EPE>"
+      )
+
+    if batch_size:
+      self.add_argument(
+          "--batch_size", "-bs", type=int, default=32,
+          help="[default: %(default)s] Batch size for training and evaluation.",
+          metavar="<BS>"
+      )
+
+
+class PerformanceParser(argparse.ArgumentParser):
+  """Default parser for specifying performance tuning arguments.
+
+  Args:
+    add_help: Create the "--help" flag. False if class instance is a parent.
+    num_parallel_calls: Create a flag to specify parallelism of data loading.
+    inter_op: Create a flag to allow specification of inter op threads.
+    intra_op: Create a flag to allow specification of intra op threads.
+  """
+
+  def __init__(self, add_help=False, num_parallel_calls=False, inter_op=False,
+               intra_op=False):
+    super(PerformanceParser, self).__init__(add_help=add_help)
+
+    if num_parallel_calls:
+      self.add_argument(
+          "--num_parallel_calls", "-npc",
+          type=int, default=5,
+          help="[default: %(default)s] The number of records that are "
+               "processed in parallel  during input processing. This can be "
+               "optimized per data set but for generally homogeneous data sets,"
+               " should be approximately the number of available CPU cores.",
+          metavar="<NPC>"
+      )
+
+    if inter_op:
+      self.add_argument(
+          "--inter_op_parallelism_threads", "-inrt",
+          type=int, default=0,
+          help="[default: %(default)s Number of inter_op_parallelism_threads to"
+               " use for CPU. See TensorFlow config.proto for details.",
+          metavar="<INRT>"
+      )
+
+    if intra_op:
+      self.add_argument(
+          "--intra_op_parallelism_threads", "-inat",
+          type=int, default=0,
+          help="[default: %(default)s Number of intra_op_parallelism_threads to"
+               " use for CPU. See TensorFlow config.proto for details.",
+          metavar="<INAT>"
+
+      )
+
+
+class DummyParser(argparse.ArgumentParser):
```

Comment:
DummyParse gives the impression that it does nothing, whereas it is actually doing something.

Reply:
maybe DefaultParser?

Reply:
Ditto on above.